File: backend/test_cicd_errors.py
"""
Test file to verify CI/CD error detection system

This file intentionally contains linter errors that should be:
1. Detected by the CI/CD pipeline
2. Captured in error logs
3. Auto-fixed by the auto-fix-linter workflow
4. Commented on the PR by comment-on-failure workflow
"""

# Linter error: unused import (will be auto-fixed)
import os
import sys
import json

# Linter error: unused variable (will be auto-fixed)
unused_variable = "This should be removed"

# Linter error: line too long (will be auto-fixed)
very_long_line_that_exceeds_the_maximum_line_length_and_should_be_wrapped_by_the_formatter_to_comply_with_pep8_standards = "test"


def test_cicd_error_detection():
    """Test function to verify error detection"""
    # Linter error: missing docstring formatting
    x=1+2  # Linter error: no spaces around operators

    # This will pass
    result = x * 2
    assert result == 6

    return result


# Linter error: multiple blank lines at end of file

File: frontend/src/testCicdErrors.ts
/**
 * Test file to verify CI/CD error detection system
 *
 * This file intentionally contains ESLint errors that should be:
 * 1. Detected by the CI/CD pipeline
 * 2. Captured in error logs
 * 3. Auto-fixed by the auto-fix-linter workflow
 * 4. Commented on the PR by comment-on-failure workflow
 */

// ESLint error: unused variable
const unusedVariable = 'This should be removed';

// ESLint error: console.log in production code
console.log('This is a test');

// ESLint error: var instead of const/let
var oldStyleVariable = 'Should use const or let';

// ESLint error: missing semicolon (if configured)
const missingSemicolon = 'test'

// ESLint error: unused import (if configured)
import { useState, useEffect } from 'react';

export function testCicdErrorDetection() {
  // ESLint error: prefer arrow function
  const myFunction = function() {
    return 'test';
  };

  // ESLint error: unnecessary return
  if (true) {
    return myFunction();
  } else {
    return 'default';
  }
}

// ESLint error: default export not allowed (if configured)
export default testCicdErrorDetection;
